src/OpenSpaceToolkit/Astrodynamics/GuidanceLaw/QLaw.cpp

```diff
@@ -193,6 +178,21 @@ Vector3d QLaw::computeThrustDirection(const Vector6d& aCOEVector, const double&
 
     const Vector3d D = -(jacobian.transpose() * derivativeMatrix).normalized();
 
+    const Vector5d deltaCOE = {
+        (aCOEVector[0] - targetCOEVector_[0]),
+        (aCOEVector[1] - targetCOEVector_[1]),
+        (aCOEVector[2] - targetCOEVector_[2]),
+        (std::acos(std::cos((aCOEVector[3] - targetCOEVector_[3])))),
+        (std::acos(std::cos((aCOEVector[4] - targetCOEVector_[4])))),
+    };
+
+    // Within tolerance of all targeted elements. No need to thrust.
+    if ((parameters_.controlWeights_.array() * deltaCOE.array().abs() <= parameters_.convergenceThresholds_.array())
+            .all())
+    {
+        return {0.0, 0.0, 0.0};
+    }
```

Comment:
Does controlWeights ensure the delta are all scaled properly and end-up in the same range?

Reply:
Yeah the control weights ensure that we are ignoring the delta between the "free" variables.